Tests/NIOTests/ChannelTests.swift

```diff
@@ -1716,4 +1716,92 @@ public class ChannelTests: XCTestCase {
             XCTFail("unexpected error: \(error)")
         }
     }
+
+    func testChannelReadsDoesNotHappenAfterRegistration() throws {
+        class ReadDoesNotHappen: ChannelInboundHandler {
+            typealias InboundIn = Any
+            private let hasRegisteredPromise: EventLoopPromise<()>
+            private let hasUnregisteredPromise: EventLoopPromise<()>
+            init(hasRegisteredPromise: EventLoopPromise<()>, hasUnregisteredPromise: EventLoopPromise<()>) {
+                self.hasRegisteredPromise = hasRegisteredPromise
+                self.hasUnregisteredPromise = hasUnregisteredPromise
+            }
+            func channelRead(ctx: ChannelHandlerContext, data: NIOAny) {
+                XCTFail("read has happened and shouldn't have")
+            }
+            func channelActive(ctx: ChannelHandlerContext) {
+                XCTFail("shouldn't have been activated")
+            }
+            func channelRegistered(ctx: ChannelHandlerContext) {
+                self.hasRegisteredPromise.succeed(result: ())
+            }
+            func channelUnregistered(ctx: ChannelHandlerContext) {
+                self.hasUnregisteredPromise.succeed(result: ())
+            }
+        }
+
+        let elg = MultiThreadedEventLoopGroup(numThreads: 1)
+        defer {
+            XCTAssertNoThrow(try elg.syncShutdownGracefully())
+        }
+        let sc = try SocketChannel(eventLoop: elg.next() as! SelectableEventLoop, protocolFamily: AF_INET)
```

Comment:
IMHO we use `PF_INET` usually, so maybe change this as well ?

Reply:
`AF_INET` is newer, more standard and overall more preferred. https://stackoverflow.com/questions/6729366/what-is-the-difference-between-af-inet-and-pf-inet-in-socket-programming